apps/studio/src/lib/db/clients/postgresql.ts

```diff
@@ -138,8 +147,33 @@ export class PostgresClient extends BasicDatabaseClient<QueryResult> {
       pool: new pg.Pool(dbConfig)
     };
 
-    //test connection
     const test = await this.conn.pool.connect()
+
+    if (this.server.config.redshiftOptions?.iamAuthenticationEnabled) {
+      this.interval = setInterval(async () => {
+        try {
+          const newPassword = await refreshTokenIfNeeded(this.server.config.redshiftOptions, this.server, 5432);
```

Comment:
@ClarLabs is this hard coded port intentional? It could presumably change no? Should it not use this.server.config.port?

Reply:
Good shout add if check